src/LabelerServer.ts

```diff
@@ -176,43 +203,30 @@ export class LabelerServer {
 	/**
 	 * Start the server.
 	 * @param port The port to listen on.
-	 * @param callback A callback to run when the server is started.
+	 * @returns address of the server
 	 */
-	start(port: number, callback: (error: Error | null, address: string) => void): void;
+	async start(port: number): Promise<string>;
 	/**
 	 * Start the server.
 	 * @param options Options for the server.
-	 * @param callback A callback to run when the server is started.
+	 * @returns address of the server
 	 */
-	start(
-		options: FastifyListenOptions,
-		callback: (error: Error | null, address: string) => void,
-	): void;
-	start(
-		portOrOptions: number | FastifyListenOptions,
-		callback: (error: Error | null, address: string) => void = () => {},
-	) {
-		if (typeof portOrOptions === "number") {
-			this.app.listen({ port: portOrOptions }, callback);
-		} else {
-			this.app.listen(portOrOptions, callback);
-		}
-	}
+	async start(options: FastifyListenOptions): Promise<string>;
+	async start(portOrOptions: number | FastifyListenOptions) {
+		// ensure the database is setup prior to listening:
+		await this.#dbInitLock;
 
-	/**
-	 * Stop the server.
-	 * @param callback A callback to run when the server is stopped.
-	 */
-	close(callback: () => void = () => {}) {
-		this.app.close(callback);
+		const options = typeof portOrOptions === "number" ? { port: portOrOptions } : portOrOptions;
+
+		return await this.app.listen(options);
 	}
 
 	/**
-	 * Alias for {@link LabelerServer#close}.
-	 * @param callback A callback to run when the server is stopped.
+	 * Stop the server & closes the database:
 	 */
-	stop(callback: () => void = () => {}) {
-		this.close(callback);
+	async stop(): Promise<void> {
+		await this.app.close();
+		this.db.close();
```

Comment:
This would mean that you couldn't emit labels after the server has stopped, but I think that's fine?